### PR TITLE
Adds parent campaign link to campaign run template 

### LINKED ISSUE #6326: Add Link of Parent Campaign to Run
#### User Story

If a campaign lead is editing a run, and then saves the run, they are brought to a page that looks like the one below. It's not helpful for them -- it doesn't show what the closed state will look like, nor does it give them any way to navigate to the page of the parent campaign. It would be great if either:

a) this page could have a link to the campaign and the campaign's closed state
b) this page could look like the published state of the run 
#### Additional Information (optional)

Right now, campaign leads who arrive at the page below are going to the Drupal search content view to find the link to the parent campaign node. :unamused: 
#### Why This Matters

There should be an easy way for them to navigate to the campaign page.
#### Relevant Screenshots and Links

https://thor.dosomething.org/us/node/6556
https://www.dosomething.org/us/node/5411/closed

image

### PR DESCRIPTION
**Pull request by chloealee**
#### What's this PR do?

Makes the parent campaign title a clickable link to return to campaign page. 
#### How should this be manually tested?
- From campaign edit page (e.g. http://dev.dosomething.org:8888/us/node/1173/edit), click on current run link. 
- This will take you to edit page of campaign run (e.g. http://dev.dosomething.org:8888/us/node/1816/edit/en). Scroll down and click "Save." 
- This will take you to campaign run edit page (e.g. http://dev.dosomething.org:8888/us/node/1816). Here you will see new template with clickable campaign title to parent campaign page. 
#### What are the relevant tickets?

Fixes #6326 

cc'ing @sbsmith86 to see the code and for all the help on this!

### CODE REVIEW

File: lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.theme.inc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * Preprocess funtions for the dosomething_campaign_run module.
+ */
+
+/**
+ * Implements hook_preprocess_node().
+ */
+function dosomething_campaign_run_preprocess_node(&$vars) {
+  if ($vars['type'] != 'campaign_run' || $vars['view_mode'] != 'full') return;
+
+  $campaign_nid = dosomething_campaign_run_get_parent_nid($vars['node']->nid);
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.theme.inc]: this value should actually already be in the `vars` array, in `field_campaigns`

File: lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.theme.inc [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  $vars['campaign_nid'] = $campaign_nid;
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.theme.inc]: so you don't need to save it as a different value here

File: lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.theme.inc [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * Preprocesses variables for campaign run page.
+ *
+ * @param array $vars
+ *  Node variables, passed from the preprocess_node.
+ * @param object $wrapper
+ *  The corresponding entity wrapper for the node in $vars. 
+ */
+function dosomething_campaign_run_preprocess_page(&$vars, &$wrapper) {  
+
+  // Use the pitch page template to the theme.
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.theme.inc]: this comment i think is wrong

File: lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.theme.inc [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * @param array $vars
+ *  Node variables, passed from the preprocess_node.
+ * @param object $wrapper
+ *  The corresponding entity wrapper for the node in $vars. 
+ */
+function dosomething_campaign_run_preprocess_page(&$vars, &$wrapper) {  
+
+  // Use the pitch page template to the theme. 
+  $vars['theme_hook_suggestions'][] = 'node__' . $vars['type'];
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.theme.inc]: shouldn't this happen automatically? it should default try to find a node template with the node type?

File: lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.theme.inc [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ * Preprocess funtions for the dosomething_campaign_run module.
+ */
+
+/**
+ * Implements hook_preprocess_node().
+ */
+function dosomething_campaign_run_preprocess_node(&$vars) {
+  if ($vars['type'] != 'campaign_run' || $vars['view_mode'] != 'full') return;
+
+  $vars['campaign_nid'] = $vars['field_campaigns'][0]['target_id'];
```
Comment (chloealee) [inline lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.theme.inc]: @angaither I think I need to assign this to this variable in order to pass it to `node--campaign_run.tpl.php`, right?

Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.theme.inc]: it should already be in the `vars` array in the `field_campaigns` field

Comment (chloealee) [inline lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.theme.inc]: ahhh yes I see - I wasn't passing it correctly before!

**Comment by angaither** [comment]
:+1:

**Pull request merged**